Sources/Helpers/OAuthHelper.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  OAuthHelper.swift
+//  OsmAnd Maps
+//
+//  Created by nnngrach on 15.05.2023.
+//  Copyright © 2023 OsmAnd. All rights reserved.
+//
+
+// OAuth version: 2.0
+// Grant type: Authorization code
+
+import SwiftUI
+import Foundation
+import AuthenticationServices
+
+@objc class OAuthHelper : NSObject {
+
+    //Override in sublcasses
+    internal class var authURL: String { return "" }
+    internal class var accessTokenURL: String { return "" }
+    internal class var clientID: String { return "" }
+    internal class var clientSecret: String { return "" }
+    internal class var redirectURI: String { return "" }
+    internal class var urlScheme: String { return "" }
+    internal class var scopes: [String] { return [] }
+    internal class var tokenSettingsKey: String { return "" }
+
+    @objc public static func getToken() -> String? {
```

Comment:
Do not declare methods as public (this is not java, please read about access levels in swift, they are very similar to obj c) For our purposes ```private``` and no keyword are enough.